pkg/appendable/index_file_csv_test.go

```diff
@@ -122,4 +162,91 @@ func TestAppendDataRowCSV(t *testing.T) {
 
 	})
 
+	// this test case fails due to the way we measure checksums
+	// since CSV's are column-based appending new columns, cause checksums and endbyteoffsets to be mismatched
+	t.Run("new index", func(t *testing.T) {
+		i, err := NewIndexFile(CSVHandler{ReadSeeker: strings.NewReader(mockCsv)})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		buf := &bytes.Buffer{}
+
+		if err := i.Serialize(buf); err != nil {
+			t.Fatal(err)
+		}
+
+		j, err := ReadIndexFile(buf, CSVHandler{ReadSeeker: strings.NewReader("header1,header2\ntest1,test2\n")})
+		if err != nil {
```

Comment:
mockCsv looks like
```sh
"header1\ntest1\n"
```

which causes every line's checksum + endbyteoffset to be mismatched